tests/test_service.py

```diff
@@ -106,6 +110,39 @@ def test_get_launch_info(self, rp_service, monkeypatch):
             verify=rp_service.verify_ssl)
         assert launch_id == {'id': 112}
 
+    def test_get_launch_info_launch_id_none(self, rp_service, monkeypatch):
+        """Test get launch information for a non started launch.
+
+        :param rp_service:  Pytest fixture that represents ReportPortalService
+                            object with mocked session.
+        :param monkeypatch: Pytest fixture to safely set/delete an attribute
+        """
+        mock_get = mock.Mock()
+        monkeypatch.setattr(rp_service.session, 'get', mock_get)
+        monkeypatch.setattr(rp_service, 'launch_id', None)
+
+        launch_info = rp_service.get_launch_info()
+        mock_get.assert_not_called()
+        assert launch_info == {}
+
+    @mock.patch('reportportal_client.service.sleep', mock.Mock())
+    @mock.patch('reportportal_client.service._get_json',
+                mock.Mock(return_value={"errorCode": 4041}))
+    def test_get_launch_info_wrong_launch_id(self, rp_service, monkeypatch):
+        """Test get launch information for a non existed launch.
+
+        :param rp_service:  Pytest fixture that represents ReportPortalService
+                            object with mocked session.
+        :param monkeypatch: Pytest fixture to safely set/delete an attribute
+        """
+        mock_get = mock.Mock()
+        monkeypatch.setattr(rp_service.session, 'get', mock_get)
+        monkeypatch.setattr(rp_service, 'launch_id', '1234')
+
+        launch_info = rp_service.get_launch_info()
+        assert mock_get.call_count == 5
+        assert launch_info == {}
```

Comment:
The second check won't be processed if the first one assert gets raised.

```suggestion
        expect(launch_info == {})
        assert_expectations()
```

Reply:
Good idea. Done.